addons/product/tests/test_variants.py

```diff
@@ -909,19 +909,103 @@ def test_name_search_dynamic_attributes(self):
         template = self.env['product.template'].create({
             'name': 'cimanyd'
         })
-        self.assertEqual(len(template.product_variant_ids), 1)
 
         template.write({
             'attribute_line_ids': [(0, False, {
                 'attribute_id': dynamic_attr.id,
                 'value_ids': [(4, dynamic_attr.value_ids[0].id, False)],
             })]
         })
-        self.assertEqual(len(template.product_variant_ids), 0)
 
         name_searched = self.env['product.template'].name_search(name='cima')
         self.assertIn(template.id, [ng[0] for ng in name_searched])
 
+    def test_dynamic_attributes_archiving(self):
+        Product = self.env['product.product']
+        ProductAttribute = self.env['product.attribute']
+        ProductAttributeValue = self.env['product.attribute.value']
+
+        # Creating attributes
+        dpa_color = ProductAttribute.create({'sequence': 1, 'name': 'color', 'create_variant': 'dynamic'})
```

Comment:
Those `pa` and `pav` abbreviations are already complicated enough, don't prefix them with `d`, we know we are on a test about dynamic.